Fix vagrant provisioning by running as unprivileged user

cjwirth opened the pull request:
After the https://github.com/IBM-Swift/Kitura/commit/c30c088d5980d7932fb97c1c57f03506ab926486 commit, which solves [#19 Move from the forked version of libdispatch (maintained by Chris) and use the official one from apple](https://github.com/IBM-Swift/Kitura/issues/19), the vagrant provisioning script stopped working. 

The main reason being that it was still pointing towards the fork of libdispatch, and the modulemap file it was pointing at no longer existing.

I went in and updated the script to work with the new version of it, and I have been able to run the sample app ("You're running Kitura") from just

~~~
> vagrant up
> vagrant ssh
> Kitura/.build/debug/KituraSample
~~~

cjwirth commented:
As a follow up question, everything that the provisioning installs (the swift development snapshot, pcre, as well as the Kitura repository itself) are all owned by `root`, rather than `vagrant`. 

Is this intentional? If so, what is the reasoning behind it?

kweinmeister commented:
Thanks for your PR! And glad you were able to get the build running successfully with Vagrant.

A couple minor things I'd recommend changing in the PR: removing the constant for the module map at the top of the file, and removing the word "patched" from the comment about downloading the patched libdispatched. Also, it wasn't intentional to install as root; that's just the default. We could do something like the following?
   config.vm.provision "shell", privileged: false, inline: <<-SHELL

cjwirth commented:
Ah, nice catches. I've went in and fixed them, so take a look.

I think it's better to not install as root, so I've changed that as well. Yep, running the script as unprivileged worked :smile:

kweinmeister commented:
Awesome! Can you follow the process in [CONTRIBUTING.md](/IBM-Swift/Kitura/blob/master/CONTRIBUTING.md), and then we can move forward?

cjwirth commented:
Hey, sorry for the delay. Was busy putting on a conference.

It looks like most of the changes were already put into the develop branch (thanks!). 
I rebased, and the only change that remained was running as an unprivileged user :sweat_smile:

cjwirth commented:
I've also made an issue https://github.com/IBM-Swift/Kitura/issues/232 in order to conform with [CONTRIBUTING.md](https://github.com/IBM-Swift/Kitura/blob/master/CONTRIBUTING.md) even though this has become a one line change :sweat_smile:

dfirsht commented:
@cjwirth Hey, sorry for the delay on getting this accepted. Have you sent in your signed CLA yet? Unfortunately we cannot accept any contributions from you until you do so.

cjwirth commented:
Yeah I sent it to the email that was listed inside the pdf. Let me know if something is off? 
![image](https://cloud.githubusercontent.com/assets/1579644/13586400/1315c6c2-e504-11e5-909c-0acb7e859679.png)

dfirsht commented:
@cjwirth No problem, it just wasn't processed yet. We can merge it now.
